lib/java/.gitignore

```diff
@@ -0,0 +1,53 @@
+# Created by https://www.toptal.com/developers/gitignore/api/java,gradle
+# Edit at https://www.toptal.com/developers/gitignore?templates=java,gradle
+
```

Comment:
This change adds a new .gitignore file in the lib/java directory. This is consistent with the recently added one for lib/kotlin. However, they do differ in some ways. And, a lot of this is redundant with the top-level .gitignore file.

I think it'd be better to remove both of these .gitignore files and add the relevant patterns into the .gitignore file at the root of the repository, rather than add this here.

Reply:
fixed in https://github.com/apache/thrift/pull/2583/commits/85c18bd6b15d45d34098e902c4337cdb7401ce2b